streaming/src/main/scala/org/apache/spark/streaming/util/FileBasedWriteAheadLog.scala

```diff
@@ -205,10 +205,12 @@ private[streaming] class FileBasedWriteAheadLog(
 
   /** Stop the manager, close any open log writer */
   def close(): Unit = synchronized {
-    if (currentLogWriter != null) {
-      currentLogWriter.close()
+    if (!executionContext.isShutdown) {
```

Comment:
@srowen, maybe I missed your point. However, wouldn't there be a change to try `shutdown` multiple times if `isShutdown` is not synchronised together? Probably, `shutdown` is idempotent itself but just wonder if it is okay to keep this as a safeguard.